tests/storage/test_registration.py

```diff
@@ -166,3 +168,101 @@ def test_3pid_inhibit_invalid_validation_session_error(self) -> None:
             ThreepidValidationError,
         )
         self.assertEqual(e.value.msg, "Validation token not found or has expired", e)
+
+
+class ApprovalRequiredRegistrationTestCase(HomeserverTestCase):
+    def default_config(self) -> JsonDict:
+        config = super().default_config()
+
+        # If there's already some config for this feature in the default config, it
+        # means we're overriding it with @override_config. In this case we don't want
+        # to do anything more with it.
+        msc3866_config = config.get("experimental_features", {}).get("msc3866")
+        if msc3866_config is not None:
+            return config
+
+        # Require approval for all new accounts.
+        config["experimental_features"] = {
+            "msc3866": {
+                "enabled": True,
+                "require_approval_for_new_accounts": True,
+            }
+        }
+        return config
+
+    def prepare(self, reactor: MemoryReactor, clock: Clock, hs: HomeServer) -> None:
+        self.store = hs.get_datastores().main
+        self.user_id = "@my-user:test"
+        self.pwhash = "{xx1}123456789"
+
+    @override_config(
+        {
+            "experimental_features": {
+                "msc3866": {
+                    "enabled": True,
+                    "require_approval_for_new_accounts": False,
+                }
+            }
+        }
+    )
+    def test_approval_not_required(self) -> None:
+        """Tests that if we don't require approval for new accounts, newly created
+        accounts are automatically marked as approved.
+        """
+        self.get_success(self.store.register_user(self.user_id, self.pwhash))
+
+        user = self.get_success(self.store.get_user_by_id(self.user_id))
+        assert user is not None
+        self.assertEqual(user["approved"], 1)
```

Comment:
this smells like it wouldn't be portable to Postgres?

Reply:
(but CI is green, so not sure why this passes actually.. When you use `UserInfo` you can just `assertTrue` though, which feels a bit better of an interface to expose)

Reply:
Turns out `1 == True` in Python :neutral_face:

Reply:
I think you want `assertIs(expr, True)`, see 

https://docs.python.org/3/library/unittest.html?highlight=asserttrue#unittest.TestCase.assertTrue

Reply:
That is: `assertTrue` ought to be called `assertTruthy`

Reply:
> Turns out `1 == True` in Python neutral_face

Wait what? I don't know how I didn't know that. I know JS is like that but I for some reason thought Python didn't do it that way.... Odd..